src/libraries/System.IO.Compression/src/System/IO/Compression/ZLibStream.cs

```diff
@@ -0,0 +1,293 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Compression
+{
+    /// <summary>Provides methods and properties used to compress and decompress streams by using the ZLib data format specification.</summary>
+    public sealed class ZLibStream : Stream
+    {
+        /// <summary>The underlying deflate stream.</summary>
+        private DeflateStream _deflateStream;
+
+        /// <summary>Initializes a new instance of the <see cref="ZLibStream"/> class by using the specified stream and compression mode.</summary>
+        /// <param name="stream">The stream to which compressed data is written or from which decompressed data is read.</param>
+        /// <param name="mode">One of the enumeration values that indicates whether to compress or decompress the stream.</param>
+        public ZLibStream(Stream stream, CompressionMode mode) : this(stream, mode, leaveOpen: false)
+        {
+        }
+
+        /// <summary>Initializes a new instance of the <see cref="ZLibStream"/> class by using the specified stream, compression mode, and whether to leave the <paramref name="stream"/> open.</summary>
+        /// <param name="stream">The stream to which compressed data is written or from which decompressed data is read.</param>
+        /// <param name="mode">One of the enumeration values that indicates whether to compress or decompress the stream.</param>
+        /// <param name="leaveOpen"><c>true</c> to leave the stream object open after disposing the <see cref="ZLibStream"/> object; otherwise, <c>false</c>.</param>
```

Comment:
It could also be fixed in the porting PR, but since we are here:
```suggestion
        /// <param name="leaveOpen"><see langword="true" /> to leave the stream object open after disposing the <see cref="ZLibStream"/> object; otherwise, <see langword="false" />.</param>
```